scripts/production/homology_dump_cleanup.py

```diff
@@ -0,0 +1,484 @@
+    It expects the following arguments:
+    
+    --homology_dumps_dir (str): Required. Root directory of the homology dumps.
+    --master_db_url (str): Required. URL of the master database.
+    --before_release (int): Required. Ensembl release cutoff for cleanup (non-inclusive).
+    --dry_run (flag): Optional. If present, performs a dry run without deleting files.
+    --log (str): Optional. Path to an optional log file to record deleted files.
+
+    Example:
+    >>> parse_args()
+    Namespace(dry_run=True, homology_dumps_dir='/path/to/homology_dumps', log=None, master_db_url='mysql://user:password@host:port/database', before_release=110)
+    """
+
+    parser = argparse.ArgumentParser(description='Homology dumps cleanup script')
+
+    parser.add_argument('--homology_dumps_dir', type=str, required=True, help='Root directory of the homology dumps.')
+    parser.add_argument('--master_db_url', type=str, required=True, help='URL of the master database.')
+    parser.add_argument('--before_release', type=int, required=True, help='Ensembl release cutoff for cleanup (non-inclusive).')
+    parser.add_argument('--dry_run', action='store_true', help='Perform a dry run without deleting files.')
+    parser.add_argument('--log', type=str, help='Optional log file to record deleted files.')
+
+    return parser.parse_args()
+          
+def main() -> None:
+    """
+    Entry point for the homology dumps cleanup script.
+
+    This function performs the following tasks:
+    1. Parses command-line arguments to get the homology dumps directory, master database URL, 
+       release cutoff, dry run option, and log file path.
+    2. Configures logging based on the provided log file path, if any.
+    3. Retrieves collections and division information from the master database using the provided URL.
+    4. Calls the cleanup_homology_dumps function to perform the cleanup based on the parsed arguments.
+
+    Command-line Arguments:
+        --homology_dumps_dir (str): The root directory of the homology dumps. Required.
+        --master_db_url (str): The URL of the master database. Required.
+        --before_release (int): The Ensembl release cutoff for cleanup (non-inclusive). Required.
+        --dry_run: Perform a dry run without deleting files. Optional.
+        --log (str): Optional log file to record deleted files.
+
+    Returns:
+        None
+
+    Example command to run this script from command line:
+    >>> ./homology_dump_cleanup.py --homology_dumps_dir /hps/nobackup/flicek/ensembl/compara/sbhurji/scripts/homology_dumps 
+    --master_db_url mysql://ensro@mysql-ens-compara-prod-5:4615/ensembl_compara_master_plants --before_release 110 --dry_run 
+    --log /hps/nobackup/flicek/ensembl/compara/sbhurji/scripts/clean.log
+    """
+
+    args = parse_args()
+    logging.basicConfig(filename=args.log, level=logging.INFO, format='%(asctime)s - %(message)s')
```

Comment:
Why is logging to file set up twice in the script?

Reply:
Yeah sorry, I missed this have updated the code now.